src/tinypeel/Peeling/PeelingIO.py

```diff
@@ -76,30 +76,39 @@ def writeGenotypes(pedigree, genoProbFunc):
     args = InputOutput.args
     if not args.no_dosage:
         writeDosages(pedigree, genoProbFunc, args.out + ".dosage.txt")
-    if args.haps:
-        writeGenoProbs(pedigree, genoProbFunc, args.out + ".haps")
-
-    if args.calling_threshold is not None:
-        for thresh in args.calling_threshold:
+    if args.phased_geno_prob:
+        writePhasedGenoProbs(pedigree, genoProbFunc, args.out + ".phased_geno_prob.txt")
+    if args.geno_prob:
+        writeGenoProbs(pedigree, genoProbFunc, args.out + ".geno_prob.txt")
+    if args.geno_threshold is not None:
+        for thresh in args.geno_threshold:
+            if thresh < 1 / 3:
+                thresh = 1 / 3
             if args.binary_call_files:
                 writeBinaryCalledGenotypes(
                     pedigree, genoProbFunc, args.out + ".called." + str(thresh), thresh
                 )
-            if not args.binary_call_files:
+            if not args.binary_call_files and args.geno:
                 writeCalledGenotypes(
-                    pedigree, genoProbFunc, args.out + ".called." + str(thresh), thresh
-                )
-
-            if args.hap:
-                writeCalledPhase(
                     pedigree,
                     genoProbFunc,
-                    args.out + ".called_phase." + str(thresh),
+                    args.out + ".geno_" + str(thresh) + ".txt",
                     thresh,
                 )
 
+            if args.hap_threshold is not None and args.hap:
```

Comment:
@AprilYUZhang probably best to change this line to  `if args.hap and args.hap_threshold is not None:` - it's easier to follow, but of course does the same thing

Reply:
I agree; I made similar changes as `if args.geno and args.geno_threshold` .